### ISSUE TITLE
NSIS installer fails with long file paths

### ISSUE THREAD

**Issue by JeeH-K** [OpenMS/OpenMS#7316]
Windows server gave an error during the build-and-test job, at the Package step. It seems like an NSIS error, and I wonder if anyone has any idea why. 
Here's the link to the failed job:
https://github.com/OpenMS/OpenMS/actions/runs/7631635650/job/20790027858

**Comment by timosachsenberg** [comment]
just to repeat this here: I saw some changes by @cbielow regarding NSIS and packaging. Maybe he can comment if his changes fixes the job.

**Comment by jpfeuffer** [comment]
Congratulations 🎉  As a long-time windows hater I can tell you that you have just reached the maximum non-unicode filepath length in Windows.😂

As a lucky winner you have won one of the following beautiful tasks:

- Consider shortening your branch name.

- Or patch NSIS one more time:
https://sourceforge.net/p/nsis/patches/256/#:~:text=When%20reading%20an%20input%20file%20using%20a%20path,characters%2C%20when%20you%20prefix%20the%20path%20with%20%3F%5C

- Or use short names in doxygen. Which was reverted before because pyopenms docstrings link back to HTMLs that need to contain the class name.

- Or somehow make doxygen not document deep classes.

- Or get rid of that trash installer software from the 90s ;)

**Comment by JeeH-K** [comment]
@jpfeuffer thank you so much for the suggestion! I tested "shortening the branch name" option, and it worked like a charm

**Comment by JeeH-K** [comment]
apparently, the the installer file has been created, but did not work... gives this error while trying to open the installer file
![image](https://github.com/OpenMS/OpenMS/assets/7207883/6a122215-ef3d-4e77-a22b-f3bb9d825b0e)

**Comment by jpfeuffer** [comment]
I think I would prefer to keep it open until it expires or someone fixes it. What do you think?

**Comment by JeeH-K** [comment]
> I think I would prefer to keep it open until it expires or someone fixes it. What do you think?

no worries! Shortening the branch name is certainly not the ultimate solution

**Comment by cbielow** [comment]
> 
>     * Or patch NSIS one more time:
>       https://sourceforge.net/p/nsis/patches/256/#:~:text=When%20reading%20an%20input%20file%20using%20a%20path,characters%2C%20when%20you%20prefix%20the%20path%20with%20%3F%5C
> 

This bug is still not fixed. Just checked (https://sourceforge.net/p/nsis/code/HEAD/tree/NSIS/trunk/Source/script.cpp#l5314 still calls https://sourceforge.net/p/nsis/code/HEAD/tree/NSIS/trunk/Source/mmap.cpp#l272).
Even if it is, there is evidence that other things need fixing as well: https://sourceforge.net/p/nsis/bugs/1304/

**Comment by jpfeuffer** [comment]
Yes. You need to patch it yourself.
I think it is to be expected that without a patch NSIS will fail, even with long paths enabled, so I wouldn't say that there necessarily are other things wrong. But yeah, not sure if this is the way to go. Obviously there is not much development going on for NSIS.

**Comment by cbielow** [comment]
using https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry#enable-long-paths-in-windows-10-version-1607-and-later it should in theory not be required to apply the patch.  (this would require NSIS to be compiled with UNICODE support -- but lets assume this was done). Because then, `CreateFileW` allows filepaths up to 32k in length.
However, https://sourceforge.net/p/nsis/bugs/1304/ talks about an issue when creating the installer. Totally different scenario.
Yet, it may well be that NSIS was build without UNICODE support, and the ANSI version was used, in which case the MAX_PATH_LEN limitation will always be a problem.

**Comment by jpfeuffer** [comment]
But our issue is also an error during creating the installer, no?

I am not sure what the last picture with the damaged installer is.

**Comment by jpfeuffer** [comment]
I agree with you. If CreateFile2 CreateDirW etc were used in NSIS source code and MAX_PATH is enabled/extended (which is the default on GHA for example) everything should work fine.
But NSIS is bugged for years so I don't know if this will ever be fixed.
I had this problem 5 years ago already.

**Comment by timosachsenberg** [comment]
close as wont fix_
